python-avd/pyavd/_eos_designs/structured_config/network_services/router_ospf.py

```diff
@@ -20,85 +23,110 @@ class RouterOspfMixin(Protocol):
         # If we have static_routes in default VRF and not EPVN, and underlay is OSPF
         # Then add redistribute static to the underlay OSPF process.
         if self._vrf_default_ipv4_static_routes["redistribute_in_underlay"] and self.shared_utils.underlay_routing_protocol in ["ospf", "ospf-ldp"]:
-            ospf_processes.append({"id": self.inputs.underlay_ospf_process_id, "redistribute": {"static": {"enabled": True}}})
-        if ospf_processes:
-            return {"process_ids": ospf_processes}
+            process = EosCliConfigGen.RouterOspf.ProcessIdsItem(id=self.inputs.underlay_ospf_process_id)
+            process.redistribute.static.enabled = True
+            self.structured_config.router_ospf.process_ids.append(process)
+
+    def _set_ospf_redistribute(
+        self, process: EosCliConfigGen.RouterOspf.ProcessIdsItem, vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem
+    ) -> None:
+        """
+        Configures OSPF route redistribution settings for the given VRF.
+
+        This method enables redistribution of BGP and connected routes into OSPF,
+        setting the associated route maps if specified.
 
-        return None
+        Args:
+            process: The OSPF process configuration object.
+            vrf: The VRF object containing OSPF redistribution settings.
+        """
+        if vrf.ospf.redistribute_bgp.enabled:
+            process.redistribute.bgp.enabled = True
+            if route_map := vrf.ospf.redistribute_bgp.route_map:
+                process.redistribute.bgp.route_map = route_map
+
+        if vrf.ospf.redistribute_connected.enabled:
+            process.redistribute.connected.enabled = True
+            if route_map := vrf.ospf.redistribute_connected.route_map:
+                process.redistribute.connected.route_map = route_map
+
+    def _set_ospf_interface(
+        self,
+        process: EosCliConfigGen.RouterOspf.ProcessIdsItem,
+        vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem,
+    ) -> None:
+        """
+        Populates the list of OSPF-enabled interfaces for the given VRF.
+
+        This method iterates through L3 interfaces and SVIs, adding those that have OSPF enabled.
+
+        Args:
+            process: The OSPF process configuration object.
+            vrf: The VRF object containing interface OSPF settings.
+        """
+        for l3_interface in vrf.l3_interfaces:
+            if l3_interface.ospf.enabled:
+                for node_index, node in enumerate(l3_interface.nodes):
+                    if node != self.shared_utils.hostname:
+                        continue
+                    process.no_passive_interfaces.append(l3_interface.interfaces[node_index])
+
+        for svi in vrf.svis:
+            if svi.ospf.enabled:
+                interface_name = f"Vlan{svi.id}"
+                process.no_passive_interfaces.append(interface_name)
+
+    def _set_ospf_vrf(
```

Comment:
I don't think this needs to be in it's own function instead of just doing it in the main function.

Reply:
I had to create a new function to resolve sonarlint issue.
Moved it back to main function.